presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java

```diff
@@ -146,10 +150,15 @@ public Node selectCurrentNode()
             List<NetworkLocation> locations = new ArrayList<>();
             for (HostAddress host : split.getAddresses()) {
                 try {
+                    if (negativeNetworkLocationCache.getIfPresent(host) != null) {
+                        continue;
+                    }
                     locations.add(networkLocationCache.get(host));
                 }
                 catch (ExecutionException | UncheckedExecutionException e) {
                     // Skip addresses we can't locate
+                    negativeNetworkLocationCache.put(host, true);
+                    log.warn(e, "Unable to locate %s. Negatively caching it", host);
```

Comment:
Maybe `Unable to determine location of host %s.  Will attempt again after in %s`... this will require making the timeout a constant.